capabilities/istio/pepr/istio-job-termination.ts

```diff
@@ -0,0 +1,69 @@
+import { Exec, KubeConfig } from "@kubernetes/client-node";
+import { Capability, Log, a } from "pepr";
+
+export const IstioJobTermination = new Capability({
+  name: "istio-job-termination",
+  description: "Ensure Istio sidecars are terminated after job completion",
+});
+
+// Use the 'When' function to create a new action
+const { When } = IstioJobTermination;
+
+// Keep track of in-progress terminations
+const inProgress: Record<string, boolean> = {};
+
+When(a.Pod)
+  .IsUpdated()
+  .WithLabel("batch.kubernetes.io/job-name")
+  .WithLabel("service.istio.io/canonical-name")
+  .Watch(async pod => {
+    const { metadata, status } = pod;
+    const { name, namespace } = metadata;
+
+    // Ensure termination isn't already in progress
+    if (inProgress[name]) {
+      return;
+    }
+
+    // Only terminate if the pod is running
+    if (status.phase == "Running") {
+      // Check if the pod has a non-istio container that has terminated
+      const canTerminate = !!status?.containerStatuses?.find(
+        ({ name, state }) => name != "istio-proxy" && state?.terminated,
```

Comment:
I don't think finding a single terminated container is sufficient here - a job could have 2, 3, ... other containers and we wouldn't want to kill the sidecar until all are terminated. This check would kill it based on a single terminated non-istio container unless I'm mistaken?

I also think we may need to check the containerstatus exit codes? If a container is terminated, but had a non-zero exit code, it could restart depending on the job restartpolicy. We could end up in a scenario where we kill the sidecar too early as a result and mess up networking for the retries.

Reply:
On multiple containers, yes I think you're right, I was distracted by a [type bug](https://github.com/defenseunicorns/pepr/pull/356) in pepr with ramda and didn't realize this was an all vs any match when I was debugging with your example (since it was only one container). 

I do think we need to make 100% sure we understand the behavior of the job controller when a multi-container pod has one or more failed condition states. Does it restart / recreate the pod? I dont use jobs enough to say confidently how that controller handles those different scenarios.

Reply:
https://kubernetes.io/docs/concepts/workloads/controllers/job/#handling-pod-and-container-failures - restart policy does not replace the pod on failure, just the individual failed container. I think that matches what I've seen with jobs in the past, only individual containers are restarted based on failures, not the whole pod.